bfloat/bfloat_test.go

```diff
@@ -0,0 +1,87 @@
+package bfloat
+
+import (
+	"math"
+	"testing"
+)
+
+func TestNewFromBits(t *testing.T) {
+	golden := []struct {
+		bits uint16
+		want float64
+	}{
+		// Special numbers.
+		// 0 00000000 0000000 = 0
+		{bits: 0, want: 0},
+		// 1 00000000 0000000 = -0
+		{bits: 0x8000, want: 1. / math.Inf(-1)},
+		// 0 11111111 0000000 = +Inf
+		{bits: 0x7f80, want: math.Inf(1)},
+		// 1 11111111 0000000 = -Inf
+		{bits: 0xff80, want: math.Inf(-1)},
+
+		// FIXME: the following comment case correctly get into nan branch but calculation of big.Float is incorrect
+		// // 0 11111111 0000001 = +NaN
+		// {bits: 0x7f81, want: math.NaN()},
+		// // 1 11111111 0000001 = -NaN
+		// {bits: 0xff81, want: -math.NaN()},
```

Comment:
The problem is the following code:

	// +-NaN
	if signbit {
		x.Neg(x)
	}
	return x, true

didn't make x became NaN

Reply:
@mewmew any idea?

Reply:
Hi @dannypsnl,

This is because `*math/big.Float` does not have a NaN representation. For this reason the `bfloat.Float.Big` method returns two values, the `*big.Float` and a boolean indicating whether the value is Not-A-Number. So, in the test case you also have to check the boolean return value (currently it is discarded with `_`).